src/Components/SingleArticle.jsx

```diff
@@ -0,0 +1,30 @@
+import { useParams } from "react-router";
+import { useEffect, useState } from "react";
+import { fetchArticleById } from "../Utils/ApiCalls";
+import { getTimeDate } from "../Utils/getTimeDate";
+
+export const SingleArticle = () => {
+  const { articleID } = useParams();
+  const [article, setArticle] = useState([]);
+
+  useEffect(() => {
+    fetchArticleById(articleID).then((article) => {
+      setArticle(article);
+    });
+  }, [articleID]);
+
+  if (article.length === 0) return <p>Loading article...</p>;
```

Comment:
Please look at using the isLoading pattern in state!
- Please also do something similar for if an error occurs (e.g someone goes to the endpoint of an article that doesn't exist.)